src/components/PostInfo/PostInfo.tsx

```diff
@@ -1,5 +1,48 @@
 import React from 'react';
 
-export const PostInfo: React.FC = () => (
-  <>Put the post here</>
-);
+import './PostInfo.scss';
+
+import { Post } from '../../types/Post';
+import { UserInfo } from '../UserInfo/UserInfo';
+import { CommentList } from '../CommentList';
+
+type Props = {
+  post: Post;
+};
+
+export const PostInfo: React.FC<Props> = ({ post }) => {
+  const {
+    title,
+    body,
+    user,
+    comments,
+  } = post;
+
+  return (
+    <>
+      {user && (
+        <div className="PostInfo">
+          <div className="PostInfo__header">
+            <h3 className="PostInfo__title">
+              {title}
+            </h3>
+
+            <p>
+              {' Posted by  '}
+
+              <UserInfo user={user} />
+            </p>
+          </div>
+
+          <p className="PostInfo__body">
+            {body}
+          </p>
+
+          {comments && (
+            <CommentList comments={comments} />
+          )}
```

Comment:
to fix tests: check if the `comments.length > 0`. If yes - render list otherwise render `no comments`:
![image](https://user-images.githubusercontent.com/86923089/214661718-b8caee79-fc0a-45e4-8c3c-f2013cfe17be.png)